### ISSUE TITLE
Issues with eu-central-1

### ISSUE THREAD

**Issue by deltamualpha** [elastic/elasticsearch-cloud-aws#165]
The documentation says it supports it, but when I try and start up a new node in Frankfurt with ES 1.4.2 and cloud-aws 2.4.1 I recieve the following error in the logs:

`1) ElasticsearchIllegalArgumentException[No automatic endpoint could be derived from region [eu-central-1]]2)`

Any guesses what's up with that?

**Comment by deltamualpha** [comment]
Aaaaand I just realized [that version's docs](https://github.com/elasticsearch/elasticsearch-cloud-aws/tree/v2.4.1/#version-241-for-elasticsearch-14) *don't* say it supports `eu-central-1`.

Is the release of the next version that does dependent on when the next major version of Elasticsearch is released?

**Comment by GlenRSmith** [comment]
I just ran into this as well. It would be nice if the added regions got added.

**Comment by mcbain** [comment]
+1

**Comment by mcbain** [comment]
IMHO one can set the 'cloud.aws.ec2.endpoint' property

{code}
if (settings.get("cloud.aws.ec2.endpoint") != null) {
            String endpoint = settings.get("cloud.aws.ec2.endpoint");
            logger.debug("using explicit ec2 endpoint [{}]", endpoint);
            client.setEndpoint(endpoint);
        } else if (settings.get("cloud.aws.region") != null) {
{code}

**Comment by eredding-rmn** [comment]
Something else is up.  When using cloud.aws.ec2.endpoint I still see a problem with auth:

```
[2015-04-21 17:34:43,106][DEBUG][discovery.ec2            ] [host] Full exception:
com.amazonaws.AmazonServiceException: AWS was not able to validate the provided access credentials (Service: AmazonEC2; Status Code: 401; Error Code: AuthFailure; Request ID: UUID)
        at com.amazonaws.http.AmazonHttpClient.handleErrorResponse(AmazonHttpClient.java:820)
        at com.amazonaws.http.AmazonHttpClient.executeHelper(AmazonHttpClient.java:439)
        at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:245)
        at com.amazonaws.services.ec2.AmazonEC2Client.invoke(AmazonEC2Client.java:8798)
        at com.amazonaws.services.ec2.AmazonEC2Client.describeInstances(AmazonEC2Client.java:4137)
        at org.elasticsearch.discovery.ec2.AwsEc2UnicastHostsProvider.buildDynamicNodes(AwsEc2UnicastHostsProvider.java:110)
        at org.elasticsearch.discovery.zen.ping.unicast.UnicastZenPing.sendPings(UnicastZenPing.java:316)
        at org.elasticsearch.discovery.zen.ping.unicast.UnicastZenPing.ping(UnicastZenPing.java:224)
        at org.elasticsearch.discovery.zen.ping.ZenPingService.ping(ZenPingService.java:146)
        at org.elasticsearch.discovery.zen.ping.ZenPingService.pingAndWait(ZenPingService.java:124)
        at org.elasticsearch.discovery.zen.ZenDiscovery.findMaster(ZenDiscovery.java:943)
        at org.elasticsearch.discovery.zen.ZenDiscovery.innerJoinCluster(ZenDiscovery.java:337)
        at org.elasticsearch.discovery.zen.ZenDiscovery.access$6000(ZenDiscovery.java:80)
        at org.elasticsearch.discovery.zen.ZenDiscovery$JoinThreadControl$1.run(ZenDiscovery.java:1320)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
        at java.lang.Thread.run(Thread.java:745)
```

**Comment by eredding-rmn** [comment]
Looks like the 2.4.x series is due for a release.  I built a snapshot off of the es-1.4 branch and the plugin worked once installed.  I did this: 

```
$ git clone https://github.com/elastic/elasticsearch-cloud-aws.git .; 
$ git fetch --all;
$ git checkout es-1.4;
$ mvn clean install
```

The resulting build is spit out at target/releases/elasticsearch-cloud-aws-2.4.2-SNAPSHOT.zip in your project directory.  Ship that sucker over to your hosts, uninstall 2.4.1, and install the 2.4.2-SNAPSHOT:

```
$ /usr/share/elasticsearch/bin/plugin -r elasticsearch/elasticsearch-cloud-aws;
$ /usr/share/elasticsearch/bin/plugin --install cloud-aws  --url file:elasticsearch-cloud-aws-2.4.2-SNAPSHOT.zip
```

**Comment by dadoonet** [comment]
Closing as this will be fixed with 2.4.2 which will be hopefully released within some days.
